deeptrack/optics.py

```diff
@@ -1,20 +1,172 @@
-""" Features for optically imaging of samples
-
-Contains features which performs physical simulations of optical devices to
-create camera images of samples.
-
-Classes
--------
-Microscope
-    Image a sample using an optical system.
-Optics
-    Abstract base optics class.
-Fluorescence
-    Optical device for fluorescenct imaging.
-Brightfield
-    Images coherently illuminated samples.
 """
+Features for Optical Imaging of Samples.
 
+This module provides classes and functionalities for simulating optical
+imaging systems, enabling the generation of realistic camera images of
+biological and physical samples. The primary goal is to offer tools for
+modeling and computing optical phenomena such as brightfield, fluorescence,
+holography, and other imaging modalities.
+
+Key Features
+------------
+- **Microscope Simulation**
+
+  The `Microscope` class acts as a high-level interface for imaging samples
+  using defined optical systems. It coordinates the interaction between the
+  sample and the optical system, ensuring seamless simulation of imaging
+  processes.
+
+- **Optical Systems**
+
+  The `Optics` class and its derived classes represent various optical
+  devices, defining core imaging properties such as resolution, magnification,
+  numerical aperture (NA), and wavelength. Subclasses like `Brightfield`,
+  `Fluorescence`, `Holography`, `Darkfield`, and `ISCAT` offer specialized
+  configurations tailored to different imaging techniques.
+
+- **Sample Illumination and Volume Simulation**
+
+  Features like `IlluminationGradient` enable realistic simulation of
+  non-uniform sample illumination, critical for generating realistic images.
+  The `_create_volume` function facilitates combining multiple scatterers
+  into a single unified volume, supporting 3D imaging.
+
+- **Integration with DeepTrack**
+
+  Full compatibility with DeepTrack's feature pipeline allows for dynamic
+  and complex simulations, incorporating physics-based models and real-time
+  adjustments to sample and imaging properties.
+
+Module Structure
```

Comment:
La sezione "Module Structure", andrebbe formattata come questa (per via di come viene quando la documentazione viene generata):

Module Structure
----------------
Classes:

- `Image`: Core class for managing array-like data and their properties.

Utility Functions:

- `def strip(element)`

    def strip(
        element: Union[Image, List, Tuple, Any],
    ) -> Any
    
    Recursively extract the underlying value from an Image object.

- `coerce(images)`

    coerce(
        images: List[Union[Image, np.ndarray]],
    ) -> List[Image]
    
    Coerce a list of images to a consistent type.

- `pad_image_to_fft(image, axes)`

    pad_image_to_fft(
        image: Union[Image, np.ndarray],
        axes: Iterable[int] = (0, 1),
    ) -> Union[Image, np.ndarray]
    
    Pads an image to optimize Fast Fourier Transform (FFT) performance.

- `maybe_cupy(array)`

    maybe_cupy(
        array: Union[np.ndarray, List, Tuple],
    ) -> Union['cupy.ndarray', np.ndarray]

    Convert an array to a CuPy array if GPU is available and enabled.